### PR TITLE
let browser generate content-type for sending form-data (fix #1122)

### LINKED ISSUE #1122: when use multer to upload file
# Describe the bug
apidoc: 0.50.2

I try to use @apiParam  or @apiBody  to  mark upload file

image
 but it show that 
image
Error: Multipart: Boundary not found

my ejs page is work:
image

# Steps to reproduce

# Information

- Version of apidoc:0.50.2
- Operating system (e.g. GNU+Linux/MacOS/Windows): Windows
- Node version: 14.17.3
- Browser (e.g. firefox, chrome): chrome 96

### PR DESCRIPTION
**Pull request by thenewvu**
these 2 commits ensure that the browser will receive a request without content-type header in the multipart case, so it has to generate a proper content-type for the request.

From jquery doc:
> When sending data to the server, use this content type. Default is "application/x-www-form-urlencoded; charset=UTF-8", which is fine for most cases. If you explicitly pass in a content-type to $.ajax(), then it is always sent to the server (even if no data is sent). As of jQuery 1.6 you can pass false to tell jQuery to not set any content type header. Note: The W3C XMLHttpRequest specification dictates that the charset is always UTF-8; specifying another charset will not force the browser to change the encoding. Note: For cross-domain requests, setting the content type to anything other than application/x-www-form-urlencoded, multipart/form-data, or text/plain will trigger the browser to send a preflight OPTIONS request to the server. https://api.jquery.com/jquery.ajax/

### CODE REVIEW

File: template/src/send_sample_request.js [hunk 1]
```diff
@@ -140,10 +140,12 @@ function sendSampleRequest (group, name, version, method) {
     }
     requestParams.data = formData;
     requestParams.processData = false;
-    // GET and DELETE methods do not need content-type
-    if (method.toLowerCase() === 'get' || method.toLowerCase() === 'delete') {
-      delete requestParams.headers['Content-Type'];
-    }
+    // With no content-type header, browser will know it need to generate a proper content type for
```
Comment (NicolasCARPi) [inline template/src/send_sample_request.js:143]: it would be `needs` and `content-type`

File: template/src/send_sample_request.js [hunk 2]
```diff
@@ -140,10 +140,12 @@ function sendSampleRequest (group, name, version, method) {
...
+    // the form data when sending it. Fix #1122
```
Comment (NicolasCARPi) [inline template/src/send_sample_request.js:144]: but how does the browser knows which content-type to use? Did you test this on multiple browsers?

Comment (thenewvu) [reply template/src/send_sample_request.js:144]: I tested on Chrome, and have tested on more browsers:
* firefox: multipart/form-data; boundary=---------------------------351499510217054383143954532765
* chrome: multipart/form-data; boundary=----WebKitFormBoundary98mmbfc7QnxEFKEe
* safari: multipart/form-data; boundary=----WebKitFormBoundaryVmoHCXIrZUuYVqry

I could not find a spec say how browser resolve the content-type, but according to jQuery document, it says:
> For cross-domain requests, setting the content type to anything other than application/x-www-form-urlencoded, multipart/form-data, or text/plain will trigger the browser to send a preflight OPTIONS request to the server. https://api.jquery.com/jquery.ajax/

File: template/src/send_sample_request.js (hunk 1 above)
Comment (thenewvu) [reply template/src/send_sample_request.js]: fixed

**Comment by NicolasCARPi** [comment]
Thank you!

**Pull request merged**